src/extensions/default/HealthData/HealthDataManager.js

```diff
@@ -69,113 +69,140 @@ define(function (require, exports, module) {
-                                    } else {
-                                        // For existing user, we will still cache
-                                        // the older uuid, so that we can improve
-                                        // our reporting in terms of figuring out
-                                        // the new users accurately.
-                                        olderUuid = userUuid;
-                                        userUuid  = generatedUuid;
-                                    }
-
-                                    PreferencesManager.setViewState("UUID", userUuid);
-                                    PreferencesManager.setViewState("OlderUUID", olderUuid);
-
-                                    oneTimeHealthData.uuid      = userUuid;
-                                    oneTimeHealthData.olderuuid = olderUuid;
-                                    return result.resolve(oneTimeHealthData);
-                                });
-                            } else {
-                                // Probably running on older shell, in which case we will
-                                // assign the same uuid to olderuuid.
-                                if (!userUuid) {
-                                    oneTimeHealthData.uuid = oneTimeHealthData.olderuuid = uuid.v4();
-                                } else {
-                                    oneTimeHealthData.olderuuid = userUuid;
-                                }
-
-                                PreferencesManager.setViewState("UUID",      oneTimeHealthData.uuid);
-                                PreferencesManager.setViewState("OlderUUID", oneTimeHealthData.olderuuid);
-                                return result.resolve(oneTimeHealthData);
-                            }
-                        }
+                        getSetUserID().done(function (userGuids) {
+                            oneTimeHealthData.uuid  = userGuids.userUuid;
+                            oneTimeHealthData.olderuuid = userGuids.olderUuid;
+                            result.resolve(oneTimeHealthData);
+                        });
                     });
 
             });
         return result.promise();
     }
 
+     /**
+     * will check uuid and older uuid exists or not
+     * set if not present
+     */
+
+    function getSetUserID() {
+
+        var result = new $.Deferred(),
+            userUuid = PreferencesManager.getViewState("UUID"),
+            olderUuid = PreferencesManager.getViewState("OlderUUID"),
+            userGuids = {};
+
+
+        if (userUuid && olderUuid) {
+            userGuids.userUuid      = userUuid;
+            userGuids.olderUuid = olderUuid;
+
+            result.resolve(userGuids);
+        } else {
+
+            // So we are going to get the Machine hash in either of the cases.
+            if (appshell.app.getMachineHash) {
+                appshell.app.getMachineHash(function (err, macHash) {
```

Comment:
This looks like machine fingerprinting in health reports.  Why is this change made? This looks like it might affect the 'anonymous' part of our health data metrics. UUIDs were deliberately made random to prevent traceability.

Reply:
@abose This functionality was already baked into brackets. I just moved it to a function so that can be called from multiple places.

Reply:
ah! got it. It just occurred to me that this is machine fingerprinting then. That is not a recommended practice for anonymous logging services. But the legality of that in desktop software might be different to that of hosted services. I think we should consider just using the randomly generated UUID here instead of hardwareID hashes that can uniquely identify a machine.